src/components/Slider/Slider.tsx

```diff
@@ -19,49 +24,74 @@ export type SliderProps = {
     items: TextItem[] | IconItem[];
     activeItemId: string;
     onChange: (id: string) => void;
+    ariaLabel?: string;
 };
 
 const isIconItem = (item: TextItem | IconItem): item is IconItem => (item as IconItem).icon !== undefined;
 
-export default function Slider({ items, activeItemId, onChange }: SliderProps): ReactElement<SliderProps> {
-    /*eslint-disable jsx-a11y/click-events-have-key-events,jsx-a11y/no-noninteractive-element-interactions,jsx-a11y/role-supports-aria-props */
+export default function Slider({
+    items,
+    activeItemId,
+    onChange,
+    ariaLabel = "Slider",
+}: SliderProps): ReactElement<SliderProps> {
+    const groupProps = { onChange, value: activeItemId, label: ariaLabel };
+    const radioGroupState = useRadioGroupState(groupProps);
+    const { radioGroupProps } = useRadioGroup(groupProps, radioGroupState);
+    const { isFocusVisible, focusProps } = useFocusRing();
+    const ref = useRef(null);
+
     return (
         <AnimateSharedLayout>
             <ul
                 data-test-id="slider"
                 className="w-full grid grid-flow-col auto-cols-fr justify-evenly p-0 border border-black-20 m-0 bg-black-0 rounded font-sans text-s list-none"
+                {...radioGroupProps}
             >
                 {items.map((item) => {
                     const isActive = item.id === activeItemId;
+                    const radioProps = {
+                        value: item.id,
+                        "aria-label": isIconItem(item) ? item.ariaLabel : item.name,
```

Comment:
since `item.ariaLabel` is optional should we use a fallback like `item ${index}`?

Reply:
For icon items it's not optional (only for the slider component) and for text items the name always has to be set as well, so there shouldn't be a case where a fallback is necessary.